File: 7-kyu/chicken-sexing/javascript/solution.js
const correctness = (bobsDecisions, expertDecisions) => {
  let ret = 0;
  for (let i = 0; i < bobsDecisions.length; i++) {
    if (bobsDecisions[i] === expertDecisions[i]) {
      ret += 1;
    } else if (bobsDecisions[i] === '?' || expertDecisions[i] === '?') {
      ret += 0.5;
    }
  }
  return ret;
};

File: 7-kyu/chicken-sexing/python/solution.py
def score(lhs, rhs):
    if lhs == rhs:
        return 1
    elif rhs == '?' or lhs == '?':
        return 0.5
    return 0

def correctness(bobs_decisions, expert_decisions): 
    return sum(score(lhs, rhs) for lhs, rhs in zip(bobs_decisions, expert_decisions))
